jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -211,42 +212,91 @@ def get_regions(self, region_type=None, list_of_subset_labels=None,
+        >>> type(plg.get_regions()['Subset 1'])
+        <class 'regions.shapes.circle.CirclePixelRegion'>
+        >>> type(plg.get_regions(wrt_data='NDData[DATA]')['Subset 1'])
+        <class 'regions.shapes.circle.CircleSkyRegion'>
+        >>> imviz.app.delete_subsets()
+        >>> imviz.link_data(align_by='wcs')
+        >>> plg.import_region(CirclePixelRegion(center=PixCoord(x=1163.618408203125, y=1433.47998046875), radius=141.28575134277344))  # noqa E501
+        >>> type(plg.get_regions()['Subset 2'])
+        <class 'regions.shapes.circle.CircleSkyRegion'>
+        >>> type(plg.get_regions(wrt_data='NDData[DATA]')['Subset 2'])
+        <class 'regions.shapes.circle.CirclePixelRegion'>
+
+        >>> cubeviz = Cubeviz()
+        >>> cubeviz.load_data(getfixture('spectrum1d_cube'))
+        >>> plg = cubeviz.plugins['Subset Tools']
+        >>> plg.import_region(CirclePixelRegion(center=PixCoord(x=24.27156066879736, y=22.183517455582475), radius=4.7523674964904785))  # noqa E501
+        >>> type(plg.get_regions()['Subset 1'])
+        <class 'regions.shapes.circle.CircleSkyRegion'>
+        >>> type(plg.get_regions(wrt_data='Unknown spectrum object[FLUX]')['Subset 1'])
+        <class 'regions.shapes.circle.CirclePixelRegion'>
         """
+        if return_sky_region is not None:
+            raise ValueError('return_sky_region no longer used, use wrt_data instead')
 
         if region_type is not None:
             region_type = region_type.lower()
             if region_type not in ['spectral', 'spatial']:
                 raise ValueError("`region_type` must be 'spectral', 'spatial', or None for any.")
+            elif region_type == 'spectral' and wrt_data:
+                raise ValueError('Unable to retrieve SkyRegion objects for spectral subsets')
             if ((self.config == 'imviz' and region_type == 'spectral') or
                (self.config == 'specviz' and region_type == 'spatial')):
-                raise ValueError(f"No {region_type} subests in {self.config}.")
+                raise ValueError(f"No {region_type} subsets in {self.config}.")
             region_type = [region_type]
 
         else:  # determine subset return type(s) by config, if not specified
             region_type = {'imviz': ['spatial'],
                            'specviz': ['spectral']}.get(self.config, ['spatial', 'spectral'])
 
-        sky_region_check = ((self.app._align_by == 'wcs' or self.config == 'cubeviz') and
-                            return_sky_region is None or return_sky_region)
-        reg_type = 'sky_region' if sky_region_check else 'region'
+        if isinstance(wrt_data, str) and wrt_data not in self.data_collection:
+            raise ValueError(f'{wrt_data} is not data in {self.data_collection}')
+
+        if ((self.app._align_by == 'wcs' and wrt_data is None) or
+                ((self.app._align_by == 'pixels' and self.config != 'cubeviz') and wrt_data) or
```

Comment:
is this line just needed because cubeviz does not set `app._align_by`?  Maybe we should have a follow-up ticket to have cubeviz set that (at least until we have support for the orientation plugin in cubeviz or the deconfigged instance)?